app/jobs/delete_host.rb

```diff
@@ -3,15 +3,12 @@
 # Job meant to delete hosts and associated objects asynchronously
 class DeleteHost
   include Sidekiq::Worker
-  include RecordNotFound
 
   def perform(message)
     Host.transaction do
-      Sidekiq.logger.info("Deleting rule results for host #{message['id']}")
-      RuleResult.where(host_id: message['id']).delete_all
-      rescue_not_found do
-        Sidekiq.logger.info("Destroying host #{message['id']}")
-        Host.find_by!(id: message['id']).destroy
+      Sidekiq.logger.info("Deleting related records for host #{message['id']}")
+      [RuleResult, TestResult, ProfileHost, PolicyHost].each do |model|
+        model.where(host_id: message['id']).delete_all
       end
     end
   end
```

Comment:
This PR no longer removes the DeleteHost job. It's repurposed to remove any host relationships. In addition, I've added a rake task that we can either schedule to run, say, once per week (CronJob) or have it run as part of the deployment (Job).